src/signal_scope/plotwidget.cpp

```diff
@@ -305,6 +348,15 @@ void PlotWidget::onResetYAxisScale()
   {
     area = QRectF(-1, -1, 2, 2);
   }
+  else
+  {
+      double marginMax = 1.1;
+      double marginMin = 1.1;
+      if (area.y()+area.height() < 0.0) marginMax = 0.9;
+      if (area.y() > 0.0) marginMin = 0.9;
+      area.setY(area.y()*marginMin);
+      area.setHeight(area.height()*marginMax);
```

Comment:
So the previous code (the one you found a simpler way to write) increases the bounds if a signal is constant. If after a while a signal becomes constant the area won't be valid. That means the area.united() operation won't happen, and that signal may be left out the plot. So I increased the bounds if `signalBounds.top() == signalBounds.bottom()`. But now that you asked, and I'm going over the code again, I'm doing practically the same thing after in this piece of code to make a margin on the plot. Just that (as you suggested a few comments ago) the margin should be added on the plot, and not on each signal.

I'm going to rewrite this in a simpler way.